packages/core/src/Cardano/Address/PaymentAddress.ts

```diff
@@ -75,10 +75,10 @@ export const inputsWithAddresses = (tx: HydratedTx, ownAddresses: PaymentAddress
  * @param txIn transaction input to resolve address from
  * @returns input owner address
  */
-export type ResolveInputAddress = (txIn: TxIn) => Promise<PaymentAddress | null>;
+export type ResolveInput = (txIn: TxIn) => Promise<{ address: PaymentAddress; value: Value } | null>;
```

Comment:
Any reason not to resolve the entire output?
```suggestion
export type ResolveInput = (txIn: TxIn) => Promise<TxOut | null>;
```
Either way, would be great to export a type that this resolves.

Reply:
> Any reason not to resolve the entire output?

Good point. I recall having this discussion when implementing the feature originally, but we opted for just the address to meet the immediate demands. Now that we're resolving an object, it makes sense to return the whole `txOut`

Reply:
https://github.com/input-output-hk/cardano-js-sdk/pull/628/commits/1159f9fd69adafc1a45fa237af1b53e223aaac43